src/main/java/com/amazon/ion/impl/bin/IonManagedWriter_1_1.kt

```diff
@@ -0,0 +1,349 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+package com.amazon.ion.impl.bin
+
+import com.amazon.ion.*
+import com.amazon.ion.impl.*
+import com.amazon.ion.impl.bin.DelimitedContainerStrategy.*
+import com.amazon.ion.impl.bin.SymbolInliningStrategy.*
+import com.amazon.ion.system.*
+import java.io.OutputStream
+import java.math.BigDecimal
+import java.math.BigInteger
+
+/**
+ * A managed writer for Ion 1.1 that is generic over whether the raw encoding is text or binary.
+ *
+ * TODO:
+ *  - Handling of shared symbol tables
+ *  - Proper handling of user-supplied symbol tables
+ *  - Auto-flush (for binary and text)
+ *
+ * TODO: What package does this really belong in?
+ *       See also [ManagedWriterOptions], [SymbolInliningStrategy], and [DelimitedContainerStrategy].
+ */
+internal class IonManagedWriter_1_1(
+    val userData: IonRawWriter_1_1,
+    val systemData: IonRawWriter_1_1,
+    /** Must be the output stream for the raw writer and the encoding directive writer */
+    val output: OutputStream,
```

Comment:
This is not actually necessary. It looks like the only thing this class does with it is call `close()`. I'll add it to my list of clean up tasks.